src/pointer.rs

```diff
@@ -906,18 +1222,54 @@ mod tests {
         ];
         for (input, expected) in tests {
             let actual = Pointer::parse(input).map(Pointer::as_str);
-            assert_eq!(
-                actual, expected,
-                "pointer parsing failed to meet expectations
-                \ninput: {input}
-                \nexpected:\n{expected:#?}
-                \nactual:\n{actual:#?}",
-            );
+            assert_eq!(actual, expected);
         }
     }
 
     #[test]
-    fn test_push_pop_back() {
+    fn parse_error_offsets() {
+        let err = Pointer::parse("/foo/invalid~encoding").unwrap_err();
+        assert_eq!(err.pointer_offset(), 4);
+        assert_eq!(err.source_offset(), 8);
+        assert_eq!(err.complete_offset(), 12);
+
+        let err = Pointer::parse("invalid~encoding").unwrap_err();
+        assert_eq!(err.pointer_offset(), 0);
+        assert_eq!(err.source_offset(), 0);
+
+        let err = Pointer::parse("no-leading/slash").unwrap_err();
+        assert!(err.source().is_none());
+    }
+
+    #[test]
+    #[cfg(feature = "std")]
+    fn parse_error_source() {
+        use std::error::Error;
+        let err = Pointer::parse("/foo/invalid~encoding").unwrap_err();
+        assert!(err.source().is_some());
+        let source = err.source().unwrap();
+        assert!(source.is::<InvalidEncodingError>());
+
+        let err = Pointer::parse("no-leading/slash").unwrap_err();
+        assert!(err.source().is_none());
+    }
+
+    #[test]
+    fn as_pointer() {
```

Comment:
This name doesn't match the test very well

Reply:
Agreed, renamed to `pointerbuf_as_pointer_returns_pointer`